csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/TrainingTest.cs

```diff
@@ -0,0 +1,287 @@
+
+
+                //var trainingSession = new InferenceSession(trainingPath);
+                //var trainingSession = new TrainingSession(state, trainingPath, evalPath, "");
+                var trainingSession = new TrainingSession(state, trainingPath);
+
+                trainingSession.Dispose();
+            }
+        }
+
+        [Fact(DisplayName = "TestTrainingSessionTrainStep")]
+        public void TestTrainingSessionTrainStep()
+        {
+            using (var state = new CheckpointState())
+            {
+                Assert.NotNull(state);
+                string checkpointPath = Path.Combine(Directory.GetCurrentDirectory(), "checkpoint.ckpt");
+                state.LoadCheckpoint(checkpointPath);
+                string trainingPath = Path.Combine(Directory.GetCurrentDirectory(), "training_model.onnx");
+
+                var trainingSession = new TrainingSession(state, trainingPath);
+
+                float[] expectedOutput = TestDataLoader.LoadTensorFromFile("loss_1.out");
+                var expectedOutputDimensions = new int[] { 1 };
+                float[] input = TestDataLoader.LoadTensorFromFile("input-0.in");
+                Int32[] labels = { 1, 1 };
+
+                // Run inference with pinned inputs and pinned outputs
+                using (DisposableListTest<FixedBufferOnnxValue> pinnedInputs = new DisposableListTest<FixedBufferOnnxValue>(),
+                                                            pinnedOutputs = new DisposableListTest<FixedBufferOnnxValue>())
+                {
+                    var memInfo = OrtMemoryInfo.DefaultInstance; // CPU
+
+                    // Create inputs
+                    long[] inputShape = { 2, 784 };
+                    var byteSize = inputShape.Aggregate(1L, (a, b) => a * b) * sizeof(float);
+                    pinnedInputs.Add(FixedBufferOnnxValue.CreateFromMemory<float>(memInfo, input,
+                        TensorElementType.Float, inputShape, byteSize));
+
+                    long[] labelsShape = { 2 };
+                    byteSize = labelsShape.Aggregate(1L, (a, b) => a * b) * sizeof(Int32);
+                    pinnedInputs.Add(FixedBufferOnnxValue.CreateFromMemory<Int32>(memInfo, labels,
+                        TensorElementType.Int32, labelsShape, byteSize));
+
+
+                    // Prepare output buffer
+                    long[] outputShape = { };
+                    byteSize = outputShape.Aggregate(1L, (a, b) => a * b) * sizeof(float);
+                    float[] outputBuffer = new float[expectedOutput.Length];
+                    pinnedOutputs.Add(FixedBufferOnnxValue.CreateFromMemory<float>(memInfo, outputBuffer,
+                        TensorElementType.Float, outputShape, byteSize));
+
+                    trainingSession.TrainStep(pinnedInputs, pinnedOutputs);
+                    Assert.Equal(expectedOutput, outputBuffer, new FloatComparer());
+                }
+
+                trainingSession.Dispose();
```

Comment:
>trainingSession [](http://example.com/codeflow?start=16&length=15)

This does not run on exception.  Use using clause or add to a disposable list. People use it as examples.